### ISSUE TITLE
zsync files are not being generated

### ISSUE THREAD

**Issue by Samueru-sama** [AppImage/appimagetool#84]
Looks like the static zsync has a problem because now everything that uses this appimagetool has missing zsync files. 

Examples: 

https://github.com/ivan-hc/Steam-appimage/releases/tag/continuous
https://github.com/pkgforge-dev/rofi-AppImage/releases/tag/405f523

Back in [here](https://github.com/AppImage/appimagetool/pull/47#issuecomment-2451558006) I commented that the static zsync that probono has wasn't working, appimagetool would just say that everything went right without the zsync file being made.

**Comment by brunvonlope** [comment]
Can confirm: https://gitlab.gnome.org/GNOME/gimp/-/jobs/4553101

**Comment by Samueru-sama** [comment]
Looking at the build script, I don't see anything weird besides this [patch](https://github.com/AppImage/appimagetool/blob/5332fe1890cbd787c29dfd4536f168f62b6eedb3/ci/install-static-zsyncmake.sh#L28)?

**Comment by TheAssassin** [comment]
@brunvonlope it looks like you're using some other appimagetool. I can't see an appimagetool log.

**Comment by brunvonlope** [comment]
@TheAssassin The appimagetool log is omitted and saved as a file: https://gitlab.gnome.org/GNOME/gimp/-/jobs/4553101/artifacts/external_file/appimagetool.log

**Comment by Samueru-sama** [comment]
[Another case](https://github.com/srevinsaju/Brave-AppImage/releases)

Brave AppImage Nightly v1.75.84 has the zsync file but the following releases do not.

**Comment by TheAssassin** [comment]
Investigating already.

**Comment by TheAssassin** [comment]
The contained zsyncmake works just fine. Must be the way we call it.

**Comment by Samueru-sama** [comment]
> The contained zsyncmake works just fine. Must be the way we call it.

That's odd because it used to work when the zsync from the runner was used instead, and back in the PR I tried to use the static zsyncmake binary on alpine by dropping it in PATH since the alpine repos don't have zsync and it didn't work, I had the same issue.

**Comment by TheAssassin** [comment]
I agree it is very odd. I cannot reproduce the issue when running a local build either. Could you please run a job with `export VERBOSE=1` to get more output from appimagetool?

**Comment by Samueru-sama** [comment]
> I agree it is very odd. I cannot reproduce the issue when running a local build either. Could you please run a job with `export VERBOSE=1` to get more output from appimagetool?

https://github.com/Samueru-sama/rofi-AppImage-test/actions/runs/12247065805/job/34164200494

I don't see anything out of the ordinary, other than the `.zsync` file is missing.

Zen-browser will run into this issue when they make their next release as well, so I think right now what is best is to remove the bundled zsync since this breaks all current workflows that use this appimagetool.

**Comment by TheAssassin** [comment]
Sorry, my bad, you need to run with `--verbose`. We need to introduce an environment variable for it, I suppose.

**Comment by Samueru-sama** [comment]
> Sorry, my bad, you need to run with `--verbose`. We need to introduce an environment variable for it, I suppose.

https://github.com/Samueru-sama/rofi-AppImage-test/actions/runs/12247228454/job/34164668295

**Comment by TheAssassin** [comment]
It looks like zsyncmake definitely was run. Could be a cwd issue, but I cannot reproduce it in my local dev build at the moment.

**Comment by Samueru-sama** [comment]
> It looks like zsyncmake definitely was run. Could be a cwd issue, but I cannot reproduce it in my local dev build at the moment.

I don't think it is a cwd issue since it didn't happen before with the runner zsync. 

The Zen appimage workflow will explicitly try to [upload the .zsync file](https://github.com/zen-browser/desktop/blob/dev/.github/workflows/build.yml#L389-L393) instead of uploading the files together with globs like others, [so this is going to fail](https://github.com/zen-browser/desktop/actions/runs/12246343549)

**Comment by TheAssassin** [comment]
Looking at your log, we see the following excerpt:

```
Running zsyncmake process: '/tmp/appimage_extracted_39b45ea7ed6dff1a6cccf3ea17fbd40d/usr/bin/zsyncmake' '/__w/rofi-AppImage-test/rofi-AppImage-test/rofi/rofi-405f523-anylinux-x86_64.AppImage' '-u' 'rofi-405f523-anylinux-x86_64.AppImage' 
zsync: 0.6.2
Blocksize: 2048
Length: 0
Hash-Lengths: 1,2,3
SHA-1: da39a3ee5e6b4b0d3255bfef95601890afd80709
```

This is the zsync file's header. What is really odd is the `Length: 0`. Also, I wouldn't expect to see the zsync header at all, zsync doesn't usually print it. I'm a bit lost currently.

For the record, using the statically linked zsyncmake, it looks like that doesn't produce a `.zsync` file either...

**Comment by TheAssassin** [comment]
Workaround implemented in #85. Backed up old binaries in https://github.com/AppImage/appimagetool/releases/tag/broken-zsyncmake.

**Comment by Samueru-sama** [comment]
Omg the timing:

![image](https://github.com/user-attachments/assets/287a8f5f-c6c9-44aa-9ea1-6ef858ba2b5c)
![image](https://github.com/user-attachments/assets/71154454-2a72-471d-a78e-f33f21e5a278)

Turns out Zen was running two jobs, one downloaded the appimagetool before #85 and another after and it resulted in that! Now the guy is going to be left wondering why one job randomly failed lol.

**Comment by brunvonlope** [comment]
This issue can be closed due to #93
